### ISSUE TITLE
False positive while code return wrapped pointer

### ISSUE THREAD

**Issue by master-q** [facebook/infer#499]
False positive while code return wrapped pointer. See following code:

https://github.com/master-q/practice-infer/tree/b0d810f3a326dda8d32ea0770e577a434b14635f/wrap

The code causes following result on Infer:

```
$ make verify
infer -- gcc `pkg-config glib-2.0 --cflags` -c main.c
Infer version v0.9.4-e757d92
Copyright 2009 - present Facebook. All Rights Reserved.
Starting analysis (Infer version v0.9.4-e757d92)

legend:
  "F" analyzing a file
  "." analyzing a procedure

F....
Analyzed 1 file

Found 1 issue

main.c:15: error: MEMORY_LEAK
   memory dynamically allocated to new_p by call to g_malloc() at line 14, column 15 is not reachable after line 15, column 2
  13.   struct foo* create(void) {
  14.           int *new_p = g_malloc(sizeof(int));
  15. >         return wrap(new_p);
  16.   }
  17.   

Summary of the reports

  MEMORY_LEAK: 1
```

But the `new_p` is freed by `delete()` function.

**Comment by keigoi** [comment]
Here Infer could not infer specs for wrap and create since the calls g_malloc(sizeof(...)) are causing PRECONDITION_NOT_MET. Maybe sizeof(...) is evaluated to 0, as mentioned by @ddino in #497. Replacing g_malloc with malloc removed those FPs (see code below). 

```c
include <stdlib.h>

struct foo {
  int *p;
};

struct foo* wrap(int *p) {
  struct foo* f = malloc(sizeof(struct foo));
  if (!f) exit(1);
  f->p = p;
  return f;
}

struct foo* create(void) {
  int *new_p = malloc(sizeof(int));
  if (!new_p) exit(1);
  return wrap(new_p);
}

void delete(struct foo *f) {
  free(f->p);
  free(f);
}

int main() {
  struct foo *f;
  f = create();
  delete(f);
}
```

**Comment by ddino** [comment]
I'm trying to fix the problem with g_malloc in #497, hopefully we can fix both problem. 
At the moment not clear if both problems comes from  the same issue.

**Comment by jvillard** [comment]
Looks like this is no longer an issue. With infer 0.15.0 I get a memory leak, but in `main()`, which I think is expected.
